app/javascript/Pages/SingleClass.jsx

```diff
@@ -0,0 +1,220 @@
+import React from "react";
+import axios from "axios";
+import PropTypes from "prop-types";
+import { useState } from "react";
+import { Link } from "@inertiajs/inertia-react";
+import Layout from "./Layout";
+
+const SingleClass = (props) => {
+  const [click, setClick] = useState("about");
+
+  const handleClick = (newState) => setClick(newState);
+  const avatar = (image) => {
+    return `${image}?tr=,r-max`;
+  };
+
+  let requestInfo = props.request;
+
+  const buttonOnClick = () => {
+    switch (buttonState) {
+      case "pending":
+        //cancel request
+        setButtonState("loading");
+
+        axios
+          .put(`/api/match_request/${requestInfo.match_id}/`, {
+            status: "cancelled",
+            headers: window.defaultHeaders,
+          })
+          .then((response) => {
+            console.log(response.data);
+
+            setButtonState("available");
+          })
+          .catch((error) => {
+            if (error.response.data.error) {
+              alert(error.response.data.error);
+            }
+            setButtonState("pending");
+          });
+
+        break;
+
+      case "available":
+        //send request
+        setButtonState("loading");
+
+        axios
+          .post(`/api/classes/${props.class.id}/request/`, {
+            headers: window.defaultHeaders,
+          })
+          .then((response) => {
+            console.log(response.data);
+
+            requestInfo.match_id = response.data.request.id;
+            setButtonState("pending");
+          })
+          .catch((error) => {
+            if (error.response.data.error) {
+              alert(error.response.data.error);
+            }
+            setButtonState("available");
+          });
+
+        break;
+
+      default:
+        break;
+    }
+  };
+
+  //button state can be 'available', 'pending', 'accepted' or 'loading'
+  const [buttonState, setButtonState] = useState(
+    props.request.status ? props.request.status : "available"
```

Comment:
**suggestion (non-blocking)**: You can write this like `props.request.status || "available"`, it should work the same and has less repetition